converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Patient/PatientSynchronizer.cs

```diff
@@ -16,23 +22,63 @@ namespace Microsoft.Health.DicomCast.Core.Features.Worker.FhirTransaction
     public class PatientSynchronizer : IPatientSynchronizer
     {
         private readonly IEnumerable<IPatientPropertySynchronizer> _patientPropertySynchronizers;
+        private readonly DicomValidationConfiguration _dicomValidationConfiguration;
+        private readonly IExceptionStore _exceptionStore;
+        private readonly ILogger<PatientSynchronizer> _logger;
 
-        public PatientSynchronizer(IEnumerable<IPatientPropertySynchronizer> patientPropertySynchronizers)
+        public PatientSynchronizer(
+            IEnumerable<IPatientPropertySynchronizer> patientPropertySynchronizers,
+            IOptions<DicomValidationConfiguration> dicomValidationConfiguration,
+            IExceptionStore exceptionStore,
+            ILogger<PatientSynchronizer> logger)
         {
             EnsureArg.IsNotNull(patientPropertySynchronizers, nameof(patientPropertySynchronizers));
+            EnsureArg.IsNotNull(dicomValidationConfiguration, nameof(dicomValidationConfiguration));
+            EnsureArg.IsNotNull(exceptionStore, nameof(exceptionStore));
+            EnsureArg.IsNotNull(logger, nameof(logger));
 
             _patientPropertySynchronizers = patientPropertySynchronizers;
+            _dicomValidationConfiguration = dicomValidationConfiguration.Value;
+            _exceptionStore = exceptionStore;
+            _logger = logger;
         }
 
         /// <inheritdoc/>
-        public void Synchronize(DicomDataset dataset, Patient patient, bool isNewPatient)
+        public void Synchronize(FhirTransactionContext context, Patient patient, bool isNewPatient, CancellationToken cancellationToken)
         {
-            EnsureArg.IsNotNull(dataset, nameof(dataset));
+            EnsureArg.IsNotNull(context, nameof(context));
             EnsureArg.IsNotNull(patient, nameof(patient));
 
+            DicomDataset dataset = context.ChangeFeedEntry.Metadata;
+
             foreach (IPatientPropertySynchronizer patientPropertySynchronizer in _patientPropertySynchronizers)
             {
-                patientPropertySynchronizer.Synchronize(dataset, patient, isNewPatient);
+                try
+                {
+                    patientPropertySynchronizer.Synchronize(dataset, patient, isNewPatient);
+                }
+                catch (Exception ex)
```

Comment:
Is this covered with unit tests?

Reply:
I added unit tests to PatientSynchronizerTests for this method and to ImagingStudyPipelineHelperTests for some other changes, looking for other places to add testing coverage as well.

Reply:
Sounds good. Is there a specific expcetion type you catch here. Lets say you catch a OutOfMemory exception here, you don't want to handle it. You will let it bubble up.